source/lac/trilinos_sparse_matrix.cc

```diff
@@ -2310,6 +2310,15 @@ namespace TrilinosWrappers
     return ((sizeof(TrilinosScalar)+sizeof(TrilinosWrappers::types::int_type))*
             matrix->NumMyNonzeros() + sizeof(int)*local_size() + static_memory);
   }
+
+  MPI_Comm SparseMatrix::get_mpi_communicator () const
+  {
+    const Epetra_MpiComm *comm = dynamic_cast<const Epetra_MpiComm *>(&range_partitioner().Comm());
+    if (comm)
+      return comm->Comm();
+    else
+      return MPI_COMM_WORLD;
+  }
```

Comment:
There is a dynamic cast above, which will fail if the Epetra_Map is not of type Epetra_MpiComm. In this case, our mpi.h file defines MPI_COMM_WORLD as an integer set to 0, if I recall correctly.